pyamg/aggregation/smooth.py

```diff
@@ -1090,21 +1155,33 @@ def energy_prolongation_smoother(A, T, Atilde, B, Bf, Cpt_params,
         Sparsity_Pattern = sparse.csr_matrix((np.ones(T.indices.shape),
                                               T.indices, T.indptr),
                                               shape=shape)
-        
+
         AtildeCopy = Atilde.copy()
-        AtildeCopy.data[:] = 1.0
         for i in range(degree):
+            cost[0] += mat_mat_complexity(AtildeCopy, Sparsity_Pattern) / float(A.nnz)
             Sparsity_Pattern = AtildeCopy*Sparsity_Pattern
 
         # Optional filtering of sparsity pattern before smoothing
+        #   - Complexity: two passes through T for theta-filter, a sort on
+        #     each row for k-filter, adding matrices if both.
         if 'theta' in prefilter and 'k' in prefilter:
+            cost[0] += (2.0 * Sparsity_Pattern.nnz +
```

Comment:
I would move all of these cost estimates into filter_matrix_rows and truncate_rows, i.e., make 
cost = [0.0] a parameter to these two functions.